.github/workflows/build-gpu.yml

```diff
@@ -0,0 +1,49 @@
+name: Build GPU
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+  pull_request:
+  schedule:
+    # Run at the end of every day
+    - cron: "0 0 * * *"
```

Comment:
If we run it on pus to `main` and all PR. Scheduled run is probably not necessary? wdyt?

Reply:
I agree with you. This was done for consistency with [CPU nightly runs](https://github.com/pystatgen/sgkit/blob/main/.github/workflows/build.yml#L8). I don't have a strong opinion on either.